src/logic/Preferences.swift

```diff
@@ -373,6 +400,21 @@ class Preferences {
     static func jsonEncode<T>(_ value: T) -> String where T: Encodable {
         return String(data: try! JSONEncoder().encode(value), encoding: .utf8)!
     }
+    
+    static func indexToName(_ baseName: String, _ index: Int) -> String {
+        return baseName + (index == 0 ? "" : String(index + 1))
+    }
+    
+    static func nameToIndex(_ name: String) -> Int {
+        var index = name.endIndex
+        while index > name.startIndex {
+            let newIndex = name.index(before: index)
+            guard name[newIndex].isWholeNumber else { break }
+            index = newIndex
+        }
+        let suffix = name.suffix(from: index)
+        return suffix == "" ? 0 : Int(suffix)! - 1
```

Comment:
Maybe something simpler would work as well here? I'm thinking to check the last char directly like:

```swift
if name[name.endIndex].isWholeNumber {
  return  Int(name[name.endIndex])
} else {
  return 0
}
```

Reply:
It's done.